it-it/rust-it.html.markdown

```diff
@@ -0,0 +1,321 @@
+    let (a, b, c) = x;
+    println!("{} {} {}", a, b, c); // 1 ciao 3.4
+
+    // Indicizzazione
+    println!("{}", x.1); // ciao
+
+    /////////////
+    // 2. Tipi //
+    /////////////
+
+    // Strutture
+    struct Point {
+        x: i32,
+        y: i32,
+    }
+
+    let origine: Punto = Punto { x: 0, y: 0 };
+
+    // Ana struct con campi senza nome, chiamata ‘tuple struct’
+    struct Punto2(i32, i32);
+
+    let origine2 = Punto2(0, 0);
+
+    // Enum basilare, analoga a quelle del linguaggio C
+    enum Direzione {
+        Sinistra,
+        Destra,
+        Su,
+        Giu,
+    }
+
+    let su = Direzione::Su;
+
+    // Enum con campi
+    enum OpzionaleI32 {
+        UnI32(i32),
+        Niente,
+    }
+
+    let due: OpzionaleI32 = OpzionaleI32::UnI32(2);
+    let niente = OpzionaleI32::Niente;
+
+    // Generici //
+
+    struct Foo<T> { bar: T }
+
+    // Questo è definito nella libreria standard come `Option`
+    enum Opzionale<T> {
+        QualcheValore(T),
+        NessunValore,
+    }
+
+    // Metodi //
+
+    impl<T> Foo<T> {
+        // I metodi di oggetto prendono un parametro `self` esplicito
+        fn get_bar(self) -> T {
+            self.bar
+        }
+    }
+
+    let a_foo = Foo { bar: 1 };
+    println!("{}", a_foo.get_bar()); // 1
+
+    // I trait (tratti), noti come "interfacce" o "mixin" in altri linguaggi
+
+    trait Maneggiamento<T> {
+        fn maneggia(self) -> Option<T>;
+    }
+
+    impl<T> Maneggiamento<T> for Foo<T> {
+        fn maneggia(self) -> Option<T> {
+            Some(self.bar)
+        }
+    }
+
+    let altro_foo = Foo { bar: 1 };
+    println!("{:?}", altro_foo.maneggia()); // Some(1)
+
+    /////////////////////////
+    // 3. Pattern matching //
+    /////////////////////////
+
+    let foo = OpzionaleI32::UnI32(1);
+    match foo {
+        OpzionaleI32::UnI32(n) => println!("E' un i32: {}", n),
+        OpzionaleI32::Niente  => println!("Non vale niente!"),
+    }
+
+    // Pattern matching avanzato
+    struct FooBar { x: i32, y: OpzionaleI32 }
+    let bar = FooBar { x: 15, y: OpzionaleI32::UnI32(32) };
+
+    match bar {
+        FooBar { x: 0, y: OpzionaleI32::UnI32(0) } =>
+            println!("I numeri valgono zero!"),
+        FooBar { x: n, y: OpzionaleI32::UnI32(m) } if n == m =>
+            println!("I numeri sono identici"),
+        FooBar { x: n, y: OpzionaleI32::UnI32(m) } =>
+            println!("Numeri diversi: {} {}", n, m),
+        FooBar { x: _, y: OpzionaleI32::Niente } =>
+            println!("Il secondo numbero non vale niente!"),
+    }
+
+    ///////////////////////////////////////////
+    // 4. Flusso di controllo (Control flow) //
+    ///////////////////////////////////////////
+
+    // cicli/iterazione con `for`
+    let array = [1, 2, 3];
+    for i in array.iter() {
+        println!("{}", i);
+    }
+
+    // Range
+    for i in 0u32..10 {
+        print!("{} ", i);
+    }
+    println!("");
+    // prints `0 1 2 3 4 5 6 7 8 9 `
```

Comment:
I can do it if you want, but to explain how you can do it yourself:

- edit the file as you would
- add the file to the list of changes: `git add it-it/rust-it.html.markdown`
- make a new commit: `git commit -m "Fixing a word still in english"`
- push to update the github version: `git push`.

Your changes will appear here.